Dockerfile

```diff
@@ -0,0 +1,24 @@
+# --- Stage 1: Build with Rust --- (amd64)
+FROM rust:1-slim@sha256:57d415bbd61ce11e2d5f73de068103c7bd9f3188dc132c97cef4a8f62989e944 AS builder
+
+WORKDIR /app
+
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends \
+        clang \
+        libclang-dev \
+        pkg-config \
+        git && \
+    rm -rf /var/lib/apt/lists/*
+
+COPY . .
+
+RUN cargo build --release && strip target/release/zallet
+
+# --- Stage 2: Minimal runtime with distroless ---
+FROM gcr.io/distroless/cc
+
+COPY --from=builder /app/target/release/zallet /usr/local/bin/zallet
+
+USER nonroot
```

Comment:
It would be ideal to have a user created with a predefined UID:GID (we recently had a discussion about this, and created an ADR for future reference: https://github.com/ZcashFoundation/zebra/blob/main/docs/decisions/devops/0001-docker-high-uid.md). Aside from the things explained there, it also helps for visibility in case a user needs to change the permissions on their docker-compose (or Kubernetes) to match the user running the container.